plugins/pulp_rpm/plugins/importers/yum/utils.py

```diff
@@ -102,6 +111,67 @@ def strip_ns(element, uri=None):
         strip_ns(child, uri)
 
 
+def fake_xml_element(repodata_snippet):
+    """
+    Wrap a snippet of xml in a fake element so it can be coerced to an ElementTree Element
+
+    :param repodata_snippet: Snippet of XML to be turn into an ElementTree Element
+    :type  repodata_snippet: str
+
+    :return: Parsed ElementTree Element containing the parsed repodata snippet
+    :rtype:  xml.etree.ElementTree.Element
+    """
+    register_namespace('rpm', constants.RPM_NAMESPACE)
+    try:
+        # make a guess at the encoding
+        codec = 'UTF-8'
+        repodata_snippet.encode(codec)
+    except UnicodeEncodeError:
+        # best second guess we have, and it will never fail due to the nature
```

Comment:
This was carefully considered at the time, and a different conclusion was made. We can definitely re-visit, but FWIW, the createrepo_c team independently came to the same conclusion. I'm not trying to defend this behavior except to the extent that I don't think it should be changed without careful consideration, and a plan for how to accommodate users with non-conforming rpms. Failing the publish here would have a negative impact on our users, most of whom do not have the ability to "fix" the rpms they want to publish. So I appreciate you raising the concern, but I do not think this PR is the right time or place to change this behavior.